NumPy and cuNumeric behave differently in eye function with float k.

np.eye(5, k=0.0)
In Numpy, it raises TypeError: slice indices must be integers or None or have an __index__ method

cunumeric.eye(5, k=0.0)

In Cunumeric, it reports struct.error: required argument is not an integer
   cunumeric/coverage.py:110: in wrapper
       return func(*args, **kwargs)
   cunumeric/module.py:172: in eye
       result._thunk.eye(k)
   cunumeric/deferred.py:1637: in eye
       task.execute()
   ../install/lib/python3.8/site-packages/legate/core/operation.py:167: in execute
       self._context.runtime.submit(self)
   ../install/lib/python3.8/site-packages/legate/core/runtime.py:1083: in submit
       self.flush_scheduling_window()
   ../install/lib/python3.8/site-packages/legate/core/runtime.py:1076: in flush_scheduling_window
       self._schedule(ops)
   ../install/lib/python3.8/site-packages/legate/core/runtime.py:1069: in _schedule
       op.launch(strategy)
   ../install/lib/python3.8/site-packages/legate/core/operation.py:544: in launch
       result = launcher.execute_single()
   ../install/lib/python3.8/site-packages/legate/core/launcher.py:969: in execute_single
       result = self._context.dispatch_single(self.build_single_task(argbuf))
   ../install/lib/python3.8/site-packages/legate/core/launcher.py:937: in build_single_task
       argbuf.get_string(),
   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
   
       def get_string(self) -> Optional[bytes]:
           if self.string is None or self.arglen != len(self.args):
               fmtstr = "".join(self.fmt)
               assert len(fmtstr) == len(self.args) + 1
   >           self.string = struct.pack(fmtstr, *self.args)
   E           struct.error: required argument is not an integer
   
   ../install/lib/python3.8/site-packages/legate/core/_legion/util.py:407: error

--- a/cunumeric/module.py
+++ b/cunumeric/module.py
@@ -15,6 +15,7 @@
 from __future__ import annotations
 
 import math
+import operator
 import re
 from collections import Counter
 from itertools import chain
@@ -168,6 +169,7 @@ def eye(
         dtype = np.dtype(dtype)
     if M is None:
         M = N
+    k = operator.index(k)
     result = ndarray((N, M), dtype)
     result._thunk.eye(k)
     return result

